python/pyarrow/tests/test_plasma_deprecated.py

```diff
@@ -0,0 +1,34 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
Is it useful to put this in a separate test file? @jorisvandenbossche What do you think?

Reply:
I did that when doing the serialization deprecation, so it is easier to ignore all the warnings with a single line of code in the actual test file (see a bit above at https://github.com/apache/arrow/pull/14343/files#diff-d95d9a313ef902a7d822798924fac8a0d542f80c5e4ffe0c74e71a8b94b17245R35)

I think I was assuming that if you put such a filterwarnings mark on the full file, you can't catch those warnings anymore in a specific test (since they are already ignored). But not fully sure that is actually true, so maybe it does work to include those tests in the same file

Reply:
Will try it out (putting tests in same file)

Reply:
Tests in the same file seem to work: https://github.com/apache/arrow/pull/14343/commits/48bcf41b281a8d7ceb580afea3049fcb891ce896

I do get an error locally as the warning from `start_plasma_store()` seems to get ignored, can't figure out where I went wrong:
https://github.com/apache/arrow/pull/14343/commits/4e82b6b0c35f107b89aecf6cf57fb2bcb100c1fc#diff-274fcc90866f87ce0fef771796910320e35124164ebb4fcd2277e47a5a8ae3a7

Reply:
> I do get an error locally as the warning from start_plasma_store() seems to get ignored, can't figure out where I went wrong:

The warning is only raise when "entering" the context (which is automatically done one uses the `with context(): ...` construct). But so in the case of your test, you can either use a `with .. : pass` to test that this is raising the warning, or move the explicit `__enter__` call inside the block where you are testing for the warning to be raised.

Reply:
🙏